docs/architecture/adr-008-square-size-independent-message-commitments.md

```diff
@@ -0,0 +1,123 @@
+5. So far, a full node in Rollmint downloads new blocks from the DA Layer after each Celestia block, coupled tightly for syncing. With this approach, we can send blocks over the p2p Layer giving a soft-commit to full nodes. Then, they would receive the hard-commit after verifying a message inclusion proof without the need to download the blocks anymore. **P2P Blocksync** You could also achieve this by saving multiple commits for each possible square size in the block header.
+    1. P2P Blocksync allows a Rollmint full node to run a Celestia light node and not a Celestia full node.
+    2. It allows the Rollup node to continue running after Celestia halts, relying on soft commits with no data availability.
+    3. It gives the Rollup the option to run asynchronously to Celestia because you don't have to wait for new Celestia blocks/commitments of the messages.
+6. Combining P2P Blocksync and the scheme in 3, we could have multiple rollup blocks in one Celestia block. It could look like this:
+  ![multiple-blocks](./assets/multiple-blocks.png)
+7. When submitting a message to Celestia, you only sign the message over one commitment and not all square sizes.
+
+We should note that Rollups can decide to do this scheme without changing core-app apart from Number 4.
+
+## Positive Core App changes
+
+Core app simplifications after refactoring core-app:
+
+- simplifies malleation process
+- simplifies arranging the square
+- simplifies the creation of PFDs
+- **Padding Savings**
+
+We change the non-interactive default rules from this:
+
+> - Messages that span multiple rows must begin at the start of a row (this can occur if a message is longer than k shares or if the block producer decides to start a message partway through a row and it cannot fit).
+> - Messages begin at a location aligned with the largest power of 2 that is not larger than the message length or k.
+
+To this:
+*Messages start at an index that is a multiple of its `msgMinSquareSize`.*
+
+As an example, we have this diagram. Message 1 is three shares long and is followed by message 2, which is 11 shares long, so the `msgMinSquareSize` of the second message is equal to four. Therefore we have a padding of 5 shares shown in light blue. Furthermore, with the new non-interactive default rule set, a message of size 11 can start in this block at index zero and index three because they are multiples of four. Therefore, we save four shares of padding while retaining the same commitment.
+
+![Padding Savings](./assets/padding-savings.png)
+
+### Neutral
```

Comment:
[optional] this heading from the template doesn't appear necessary if we don't populate its contents

```suggestion
```